mm2src/coins/nft.rs

```diff
@@ -166,53 +245,284 @@ pub async fn update_nft(ctx: MmArc, req: UpdateNftReq) -> MmResult<(), UpdateNft
+/// If metadata info contains potential spam links, function sets `possible_spam` true.
+async fn update_transfer_meta_using_nft<T>(storage: &T, chain: &Chain, nft: &mut Nft) -> MmResult<(), UpdateNftError>
+where
+    T: NftListStorageOps + NftTransferHistoryStorageOps,
+{
+    let transfer_meta = TransferMeta::from(nft.clone());
     storage
-        .update_transfers_meta_by_token_addr_id(&nft_db.chain, transfer_meta)
+        .update_transfers_meta_by_token_addr_id(chain, transfer_meta, nft.common.possible_spam)
         .await?;
     Ok(())
 }
 
-async fn get_moralis_nft_list(ctx: &MmArc, chain: &Chain, url: &Url) -> MmResult<Vec<Nft>, GetNftInfoError> {
+/// Extracts domains from uri_meta and token_domain.
+fn gather_domains(token_domain: &Option<String>, uri_meta: &UriMeta) -> HashSet<String> {
+    let mut domains = HashSet::new();
+    if let Some(domain) = token_domain {
+        domains.insert(domain.clone());
+    }
+    if let Some(domain) = &uri_meta.image_domain {
+        domains.insert(domain.clone());
+    }
+    if let Some(domain) = &uri_meta.animation_domain {
+        domains.insert(domain.clone());
+    }
+    if let Some(domain) = &uri_meta.external_domain {
+        domains.insert(domain.clone());
+    }
+    domains
+}
+
+/// Refreshes the `possible_spam` flag based on spam results.
+async fn refresh_possible_spam<T>(
+    storage: &T,
+    chain: &Chain,
+    nft_db: &mut Nft,
+    url_antispam: &Url,
+) -> MmResult<(), UpdateNftError>
+where
+    T: NftListStorageOps + NftTransferHistoryStorageOps,
+{
+    let address_hex = eth_addr_to_hex(&nft_db.common.token_address);
+    let spam_res = send_spam_request(chain, url_antispam, address_hex.clone()).await?;
+    if let Some(true) = spam_res.result.get(&nft_db.common.token_address) {
+        nft_db.common.possible_spam = true;
+        storage
+            .update_transfer_spam_by_token_address(chain, address_hex, true)
+            .await?;
+    }
+    Ok(())
+}
+
+/// Refreshes the `possible_phishing` flag based on phishing results.
+async fn refresh_possible_phishing<T>(
+    storage: &T,
+    chain: &Chain,
+    domains: HashSet<String>,
+    nft_db: &mut Nft,
+    url_antispam: &Url,
+) -> MmResult<(), UpdateNftError>
+where
+    T: NftListStorageOps + NftTransferHistoryStorageOps,
+{
+    if !domains.is_empty() {
+        let domain_list = domains.into_iter().collect::<Vec<_>>().join(",");
+        let domain_res = send_phishing_request(url_antispam, domain_list).await?;
+        for (domain, is_phishing) in domain_res.result.into_iter() {
+            if is_phishing {
+                nft_db.possible_phishing = true;
+                storage
+                    .update_transfer_phishing_by_domain(chain, domain.clone(), is_phishing)
+                    .await?;
+                storage
+                    .update_nft_phishing_by_domain(chain, domain, is_phishing)
+                    .await?;
```

Comment:
Same question as above :)

Reply:
hmmmm this is a bit another case for me.
Well lets say that previously(during save nft process in db) in metadata we caught a phishing domain and marked nft as phishing. Then later user refreshes metadata which was changed somehow and new domains are not in our phishing domains. 

There can be the situation that there is new domain in metadata after refresh and we dont find it in our phishing db bcz database is out dated, while new domain still can be phishing.

In addition phishing sites are easier to check than a contract addresses for spam. What I mean is that if a domain or link are in a phishing database, then there is a high probability that it is really a phishing. Moreover, phishing poses a more serious threat than spam. I suggest to leave phishing flag forever once we marked nft as phishing. 
And yeah not to `send_phishing_request` in refresh function if `possible_phishing` was true.